mithril-client/src/cardano_database_client/api.rs

```diff
@@ -0,0 +1,116 @@
+use std::sync::Arc;
+
+#[cfg(feature = "fs")]
+use slog::Logger;
+
+use crate::aggregator_client::AggregatorClient;
+#[cfg(feature = "fs")]
+use crate::feedback::FeedbackSender;
+#[cfg(feature = "fs")]
+use crate::file_downloader::FileDownloader;
+
+/// HTTP client for CardanoDatabase API from the Aggregator
+pub struct CardanoDatabaseClient {
+    pub(super) aggregator_client: Arc<dyn AggregatorClient>,
+    #[cfg(feature = "fs")]
+    pub(super) http_file_downloader: Arc<dyn FileDownloader>,
+    #[cfg(feature = "fs")]
+    pub(super) feedback_sender: FeedbackSender,
+    #[cfg(feature = "fs")]
+    pub(super) logger: Logger,
+}
+
+impl CardanoDatabaseClient {
+    /// Constructs a new `CardanoDatabase`.
+    pub fn new(
+        aggregator_client: Arc<dyn AggregatorClient>,
+        #[cfg(feature = "fs")] http_file_downloader: Arc<dyn FileDownloader>,
+        #[cfg(feature = "fs")] feedback_sender: FeedbackSender,
+        #[cfg(feature = "fs")] logger: Logger,
+    ) -> Self {
+        Self {
+            aggregator_client,
+            #[cfg(feature = "fs")]
+            http_file_downloader,
+            #[cfg(feature = "fs")]
+            feedback_sender,
+            #[cfg(feature = "fs")]
+            logger: mithril_common::logging::LoggerExtensions::new_with_component_name::<Self>(
+                &logger,
+            ),
+        }
+    }
+}
+
+#[cfg(test)]
+pub(crate) mod test_dependency_injector {
+    use super::*;
+
+    use crate::{
+        aggregator_client::MockAggregatorHTTPClient,
+        feedback::FeedbackReceiver,
+        file_downloader::{FileDownloader, MockFileDownloaderBuilder},
+        test_utils,
+    };
+
+    /// Dependency injector for `CardanoDatabaseClient` for testing purposes.
+    pub(crate) struct CardanoDatabaseClientDependencyInjector {
+        http_client: MockAggregatorHTTPClient,
+        http_file_downloader: Arc<dyn FileDownloader>,
+        feedback_receivers: Vec<Arc<dyn FeedbackReceiver>>,
+    }
+
+    impl CardanoDatabaseClientDependencyInjector {
+        pub(crate) fn new() -> Self {
+            Self {
+                http_client: MockAggregatorHTTPClient::new(),
```

Comment:
It's a bit confusing to have reference to HTTP everywhere in test and mock and not in `CardanoDatabaseClient`.
It contains an attribute with the trait `AggregatorClient` and there is an HTTP implementation but it could be another one.

Reply:
Done